ste/mgr-JobPartTransferMgr.go

```diff
@@ -278,13 +284,22 @@ func (jptm *jobPartTransferMgr) Info() TransferInfo {
 	// We need to set the blockSize in such way that number of blocks per blob
 	// does not exceeds 50000 (max number of block per blob)
 	if blockSize == 0 {
-		blockSize = uint32(common.DefaultBlockBlobBlockSize)
-		for ; uint32(sourceSize/int64(blockSize)) > common.MaxNumberOfBlocksPerBlob; blockSize = 2 * blockSize {
+		blockSize = common.DefaultBlockBlobBlockSize
+		for ; uint32(sourceSize/blockSize) > common.MaxNumberOfBlocksPerBlob; blockSize = 2 * blockSize {
+			if blockSize > common.AzCopyBlockSizeThreshold {
+				/*
+				 * For a RAM usage of 0.5G/core, we would have 4G memory on typical 8 core device, meaning at a blockSize of 256M,
+				 * we can have 4 blocks in core, waiting for a disk or n/w operation. Any higher block size would *sort of*
+				 * serialize n/w and disk operations, and is better avoided.
+				 */
+				blockSize = blockSize / common.MaxNumberOfBlocksPerBlob
```

Comment:
>blockSize = blockSize / common.MaxNumberOfBlocksPerBlob [](start = 4, length = 55)

If the block is larger than the max (4GB) when we divide blob size by 50000, it means that we cannot accommodate this size and we should error out. I suggest isolating this bit of functionality into a separate function and test it with unit tests.

Reply:
This function does not have the capability to return error. Erroring out because of invalid block sizes or file size has always happened in blockBlobSenderBase.

Reply:
Moved the verification code to a different function. Added a ut to verify it.